### PR TITLE
[1537] Add option for dynamic blog content on doc pages via contextual_links frontmatter

### PR DESCRIPTION
**Pull request by mclausaudio**
The following adds the functionality to add dynamic blog content to the right-side nav's Contextual Links area.  The title of the 3 most recent blog posts of an associated tag will be rendered onto the the nav as links.

For example, in this screenshot the 3 blog posts under "Related Blog Posts" are the 3 most recent blog articles with the "collections" tag.

Screen Shot 2021-02-24 at 2 56 11 PM

**How it works**

On .md files, add the following code to it's frontmatter:

`- type: dynamic_blog
    name: "{{Title of the sub section (most often it is: Related Blog Posts}}"
    blog_tag: "{{Tag from blog}}"`

For example:

Screen Shot 2021-02-24 at 3 10 10 PM

The blog content is fetched at build time so content will be updated during each re-deployment of the site.  If there is a specific blog post you want displayed, you can still manually include that article, which is what we are doing now.  Like so:

Screen Shot 2021-02-24 at 3 23 07 PM

This dynamic blog content functionality doesn't replace our current method, but offer a hands-off alternative that renders fresher content on a more consistent basis.

Two docs with dynamic_blogs:
* http://localhost:8000/docs/running-collections/intro-to-collection-runs
* http://localhost:8000/docs/running-collections/scheduling-collection-runs

NOTE:
- You will need to `npm i` to install shelljs
- You will not be able to see the dynamic content if you don't have the environment variables saved into `.env.development` locally, an empty object will be written to `/bff-data/blog.json` instead of the recent blog data and nothing will be displayed, but the build will still succeed.

### CODE REVIEW

**Review by tristandenyer — COMMENTED** [review]
Pulled, ran and tested on a new environment without .env file (contributor / non-Postman use case.) Ran as expected, without blog post in sidebar for the two pages that have dynamic_blog flag.

**Review by tristandenyer — APPROVED** [review]
Pulled, added in .env file, ran, and checked against acceptance criteria. LGTM

File: bff.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"path":"api/ba2920d8055501453d627f56bbc5a1ee344dc45a/425e0bac7fa18d2ae80024539cfc9a6e88ba62d3","host":"https://postman-web-property-assets.s3.amazonaws.com"}
```
Comment (oswish) [inline bff.json:1]: Github is saying that it wants this file to have an empty line at the bottom;
empty-line-end-of-file

File: bff.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+const pingWebHook = require('./scripts/pingWebHook');
+const fetchBlogPosts = require('./scripts/fetchBlogPosts');
+
+const prefetch = async (dir, response) => {
+  sh.exec('mkdir -p bff-data');
+  await pingWebHook();
+  fetchBlogPosts();
+};
+
+prefetch();
```
Comment (oswish) [inline bff.js:11]: Github is saying that it wants this file to have an empty line at the bottom;
empty-line-end-of-file

File: bff.json (hunk 1 above)
Comment (oswish) [inline bff.json:1]: Github is saying that it wants this file to have an empty line at the bottom;
empty-line-end-of-file-2

File: .env.example [hunk 3]
```diff
@@ -1,8 +1,9 @@
+
 # rename this file to .env and supply the values listed below
 # also make sure they are available to the build tool (e.g. netlify)
 # warning: variables prefixed with GATSBY_ will be made available to client-side code
 # be careful not to expose sensitive data (in this case your Algolia admin key)
 
 GATSBY_ALGOLIA_APP_ID=insertValue
 GATSBY_ALGOLIA_SEARCH_KEY=insertValue
-ALGOLIA_ADMIN_KEY=insertValue
+ALGOLIA_ADMIN_KEY=insertValue
```
Comment (oswish) [inline .env.example:9]: Github is saying that it wants this file to have an empty line at the bottom;
empty-line-end-of-file-3

**Review by tristandenyer — APPROVED** [review]
LGTM

**Pull request merged**